tests/unittests/dataset_tests.py

```diff
@@ -4833,6 +4833,203 @@ def test_select_exclude_dynamic_fields(self):
         self.assertFalse(frame.has_field("field"))
         self.assertFalse(frame.predictions.detections[0].has_field("field"))
 
+    @drop_datasets
+    def test_filter_fields(self):
+        dataset = fo.Dataset()
+
+        dataset.add_samples(
+            [
+                fo.Sample(
+                    filepath="image1.jpg", ground_truth=fo.Classification()
+                ),
+                fo.Sample(
+                    filepath="image3.jpg",
+                    field_1=1,
+                    predictions=fo.Detections(
+                        detections=[fo.Detection(field_1=1)]
+                    ),
+                ),
+                fo.Sample(
+                    filepath="image4.jpg",
+                    field_2=2,
+                    predictions=fo.Detections(
+                        detections=[fo.Detection(field_2=2)]
+                    ),
+                ),
+            ]
+        )
+        dataset.add_sample_field("field_3", ftype=fo.StringField)
+
+        field_1 = dataset.get_field("field_1")
+        field_2 = dataset.get_field("field_2")
+        field_3 = dataset.get_field("field_3")
+
+        field_1.description = "this is a unique description by joe"
+        field_2.description = "hello world test123"
+
+        field_1.info = {
+            "a": 12,
+            "b": 24,
+            "c": 36,
+            "owner": "jill",
+            "test": True,
+            "d_1": {
+                "e_2": {"f_3": "oo", "g_3": {"h_4": {"i_5": {"j_6": "nope"}}}}
+            },
+        }
+        field_2.info = {
+            "list": [1, 2, 3],
+            "owner": "joe",
+            "test": True,
+            "other": "this is a unique info value",
+        }
+        field_3.info = {"one": {"two": {"three": "test123"}}}
+
+        # doesn't return anything on empty string
+        view = dataset.select_fields(meta_filter="")
+        fields = view.get_field_schema(flat=True)
+        assert "field_1" not in fields
```

Comment:
nit: can you follow the existing convention of using methods like `self.assertNotIn`?